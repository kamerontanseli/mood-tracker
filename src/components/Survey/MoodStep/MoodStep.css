.MoodStep-title {
  text-align: center;
  margin: 50px 0;
}

.MoodStep-input {
  max-width: 300px;
  margin: 0 auto;
  margin-bottom: 50px;
}

.MoodStep-controls {
  text-align: center;
}