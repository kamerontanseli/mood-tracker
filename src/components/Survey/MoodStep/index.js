import "./MoodStep.css";
import React from "react";
import Button from "../../general/Button";
import Panel from "../../general/Panel";
import Header1 from "../../general/Header1";
import Header2 from "../../general/Header2";
import RangeInput from "../../general/RangeInput";
import { getMood } from "../../../utils/string";

const MoodStep = ({ value, onChange, onNext }) => (
  <Panel>
    <div className="MoodStep">
      <div className="MoodStep-title">
        <Header1>How do you feel?</Header1>
        <Header2>{getMood(value)}</Header2>
      </div>
      <div className="MoodStep-input">
        <RangeInput max={7} min={1} value={value} onChange={onChange} />
      </div>
      <div className="MoodStep-controls">
        <Button onClick={onNext}>Continue</Button>
      </div>
    </div>
  </Panel>
);

export default MoodStep;
