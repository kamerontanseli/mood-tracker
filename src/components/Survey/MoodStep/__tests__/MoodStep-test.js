import React from "react";
import { shallow, mount } from 'enzyme';
import renderer from "react-test-renderer";
import MoodStep from "..";

describe("render", () => {
  it("should render", () => {
    const snapshot = renderer.create(
      <MoodStep value={1} />
    );
    expect(snapshot).toMatchSnapshot();
  });
  it('should call onChange when input is changed', () => {
    const spy = jest.fn()
    const wrapper = mount(<MoodStep value={1} onChange={spy} />)
    wrapper.find('input').simulate('change')
    expect(spy).toHaveBeenCalled()
  })
  it('should call onNext when [data-test="next"] is clicked', () => {
    const spy = jest.fn()
    const wrapper = mount(<MoodStep value={1} onNext={spy} />)
    wrapper.find('[data-test="next"]').first().simulate('click')
    expect(spy).toHaveBeenCalled()
  })
});
