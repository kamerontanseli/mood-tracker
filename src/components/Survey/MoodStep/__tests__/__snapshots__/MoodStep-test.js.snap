// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render 1`] = `
<div
  className="Panel"
>
  <div
    className="MoodStep"
  >
    <div
      className="MoodStep-title"
    >
      <h1
        className="Header1"
      >
        How do you feel?
      </h1>
      <h2
        className="Header2"
      >
        Terrible
      </h2>
    </div>
    <div
      className="MoodStep-input"
    >
      <input
        className="RangeInput"
        max={7}
        min={1}
        type="range"
        value={1}
      />
    </div>
    <div
      className="MoodStep-controls"
    >
      <button
        className="Button"
        data-test="next"
      >
        Continue
      </button>
    </div>
  </div>
</div>
`;
